plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/XURIBasedFileSystemAccessFactory.java

```diff
@@ -31,7 +31,7 @@
 /** Creates an {@link IFileSystemAccess file system access} that is backed by a {@link URIConverter}. */
 @SuppressWarnings("restriction")
 @Singleton
-public class XURIBasedFileSystemAccessFactory {
+class XURIBasedFileSystemAccessFactory {
```

Comment:
Isn't package accessibility a problem for people who want to customize Xtext?
If not, could we reduce accessibility to "package" for other classes in ...xtext.server.build as well?

Reply:
I tried to figure out which of the assumptions about the encapsulation of the builder package hold true. Actually I would expect that I could make all types package visible if I introduce a local Guice module for that package. So far that was not possible (e.g. due to N4jscCompiler). I made the class public again.